### PR TITLE
{Core} Add `run_az_cmd` for internal `az`-related cmd execution

### PR DESCRIPTION
**Pull request by AllyW**
**Related command**

**Description**

As discussed in cli subprocess guidance doc (https://github.com/Azure/azure-cli/pull/29681), and with `run_cmd` added in pr (https://github.com/Azure/azure-cli/pull/29745), this pr deals with `az`-related cmds execution in cli. 

After discussion with team members, we agreed on using `cli.invoke` to replace `subprocess.run("az xxxxx")` for security concerns. Other optional choices could be 1) calling sdk; 2) using atomic cmds from codegen if exists.  These are existing solutions and would be put into doc (https://github.com/Azure/azure-cli/pull/30069). 

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Core

File: src/azure-cli-core/azure/cli/core/util.py [hunk 1]
```diff
@@ -1370,3 +1370,24 @@ def run_cmd(args, *, capture_output=False, timeout=None, check=False, encoding=N
...
                           encoding=encoding, env=env)
+
+def run_az_cmd(args, out_file=None):
+    """
+    run az related during command execution
+    Args:
+        args: cmd to be executed, array of string, like `["az", "--version"]`
+        out_file: The file to send output to. file-like object
+    Returns:
+        result: cmd execution result object, containing `result`, `error`, `exit_code`
```
Comment (jiasli) [inline src/azure-cli-core/azure/cli/core/util.py:1381]: It's better to use the `:param` convention.

File: src/azure-cli-core/azure/cli/core/util.py [hunk 2]
```diff
@@ -1370,3 +1370,24 @@ def run_cmd(args, *, capture_output=False, timeout=None, check=False, encoding=N
...
+    """
+    from azure.cli.core.azclierror import ArgumentUsageError
+    if not isinstance(args, list):
+        raise ArgumentUsageError("Invalid args. run_az_cmd args must be a list")
+    if args[0] != "az":
+        raise ArgumentUsageError('Invalid cmd. run_az_cmd must be used to run `az` related cmds, '
+                                 'like ["az", "--version"]')
```
Comment (jiasli) [inline src/azure-cli-core/azure/cli/core/util.py:1388]: I think we can allow passing the command args without `az`, such as `['version']`.

File: src/azure-cli-core/azure/cli/core/util.py [hunk 3]
```diff
@@ -1370,3 +1370,22 @@ def run_cmd(args, *, capture_output=False, timeout=None, check=False, encoding=N
...
+    from azure.cli.core.azclierror import ArgumentUsageError
+    if not isinstance(args, list):
+        raise ArgumentUsageError("Invalid args. run_az_cmd args must be a list")
+    if args[0] == "az":
+        args = args[1:]
+
+    from azure.cli.core import get_default_cli
+    cli = get_default_cli()
+    cli.invoke(args, out_file=out_file)
+    return cli.result
```
Comment (AllyW) [inline src/azure-cli-core/azure/cli/core/util.py:1391]: If exception raised during cli.invoke, then no result can be fetched by caller but the exception itself.

File: src/azure-cli-core/azure/cli/core/util.py [hunk 4]
```diff
@@ -1370,3 +1370,22 @@ def run_cmd(args, *, capture_output=False, timeout=None, check=False, encoding=N
...
+    """
+    run_az_cmd would run az related cmds during command execution
+    :param args: cmd to be executed, array of string, like `["az", "version"]`, "az" is optional
+    :param out_file: The file to send output to. file-like object
+    :return: cmd execution result object, containing `result`, `error`, `exit_code`
+    """
+    from azure.cli.core.azclierror import ArgumentUsageError
+    if not isinstance(args, list):
+        raise ArgumentUsageError("Invalid args. run_az_cmd args must be a list")
+    if args[0] == "az":
```
Comment (jsntcy) [inline src/azure-cli-core/azure/cli/core/util.py:1385]: If args[0] != "az", what will happen?

Comment (AllyW) [reply src/azure-cli-core/azure/cli/core/util.py:1385]: act as @jiasli suggested, support `az version` and `version`  as `self.cmd` does in our `ScenarioTest`. Related code as below: 
```
if command.startswith('az '):
    command = command[3:]
```

**Review by kairu-ms — APPROVED** [review]

**Review by jiasli — APPROVED** [review]

**Pull request merged**